libdnf5-plugins/expired-pgp-keys/expired-pgp-keys.cpp

```diff
@@ -169,8 +169,13 @@ void ExpiredPgpKeys::process_expired_pgp_keys(const libdnf5::base::Transaction &
     }
 
     // Iterate over all installed OpenPGP keys.
-    // TODO: Respect install root.
     auto ts = rpmtsCreate();
+    auto root_dir = config.get_installroot_option().get_value();
+    if (rpmtsSetRootDir(ts, root_dir.c_str()) != 0) {
+        logger.error("Expired PGP Keys Plugin: Failed to set rpm transaction root dir \"{}\".", root_dir);
+        rpmtsFree(ts);
+        return;
```

Comment:
Shouldn't a plugin raise an exception on an error?

Reply:
I was thinking about it, but since the plugin is not mandatory and the DNF5 command in which it was executed can still succeed, I prefer to log the error rather than abort the entire process.

Reply:
Ok. It's a pitty user cannot see the failure, but that's what the current callbacks offer.